### PR TITLE
Updated exit.js to use callback from empty write instead of drain event

### LINKED ISSUE #4: 'drain' event is not being emitted, hanging process
The only place I've run into this issue is in a grunt (0.4.2) task running on my TeamCity server (a windows machine) so it may have something to do with the process.on('exit') fix not being called.

From what I can gather about the 'drain' event, I don't believe it's fired unless the stream has broken the stream._writableState.highWaterMark. So if the stream.bufferLength is not zero, but the highWaterMark hasn't been crossed, the 'drain' event will not be emitted and the process will never exit. 

`if (stream.bufferSize === 0) {`
with
`if (!stream._writableState.needDrain) {`

Seems to solve the issue for me and still preserve the task output but I'm not sure that's 100% correct either.

### PR DESCRIPTION
**Pull request by chrisgladd**
This commit fixes #4

Fix was tested in Windows 7 and 8 in cmd, powershell, and git-bash.

I also tested the original issue with a hanging build on TeamCity and the full test was reported without hanging.

### CODE REVIEW

File: lib/exit.js [hunk 1]
```diff
@@ -18,17 +18,17 @@ module.exports = function exit(exitCode, streams) {
     }
   }
   streams.forEach(function(stream) {
-    // Prevent further writing.
-    stream.write = function() {};
     // Count drained streams now, but monitor non-drained streams.
-    if (stream.bufferSize === 0) {
+    if(stream.bufferSize === 0){
```
Comment (cowboy) [inline lib/exit.js]: Can you revert this line back to the way it was before?

File: lib/exit.js [hunk 2]
```diff
@@ -18,17 +18,17 @@ module.exports = function exit(exitCode, streams) {
...
       drainCount++;
     } else {
-      stream.once('drain', function() {
+      stream.write('', 'utf-8', function() {
```
Comment (cowboy) [inline lib/exit.js:25]: Instead of writing a utf8 string, how about an empty Buffer?

File: lib/exit.js (hunk 1 above)
Comment (chrisgladd) [inline lib/exit.js]: Fixed

File: lib/exit.js (hunk 2 above)
Comment (chrisgladd) [inline lib/exit.js:25]: I tried using an empty Buffer, but I wasn't able to get things to function as expected.

**Comment by cowboy** [comment]
v0.1.2 in npm, please test ASAP!

**Pull request closed without merging**